File: Data Structures & Algorithms/permutation-string/submission-15.java
class Solution {
    public boolean checkInclusion(String s1, String s2) {
        int []map1=new int[26];
        int []map2=new int[26];
        for(char c:s1.toCharArray()){
            map1[c-'a']++;
        }
        for(int i=0;i<s2.length();i++){
            map2[s2.charAt(i)-'a']++;
            if(i>=s1.length()) map2[s2.charAt(i-s1.length())-'a']--;
            if(match(map1,map2)) return true;
        }
        return false;
    }
    public static boolean  match(int []n,int []m){
        for(int i=0;i<n.length;i++){
            if(n[i]!=m[i]) return false;
        }
        return true;
    }
}

File: Data Structures & Algorithms/is-anagram/submission-15.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        mp={}
        for c in s:
            if c not in mp:
                mp[c]=1
            else:
                mp[c]=mp[c]+1

        for c in t:
            if c not in mp:
                mp[c]=-1
            else:
                mp[c]=mp[c]-1
        
        for c in mp:
            if mp[c]!=0:
                return False
        return True